### PR TITLE
Add a flag to disable the use of the just-built compiler.

### PR DESCRIPTION
**Pull request by nadavrot**
Add a flag to disable the use of the just-built clang compiler.

### CODE REVIEW

**Comment by nadavrot** [comment]
@swift-ci Please test

File: CMakeLists.txt [hunk 1]
```diff
@@ -62,6 +62,10 @@ option(SWIFT_INCLUDE_DOCS
     "Create targets for building docs."
     TRUE)
 
+option(SWIFT_HOST_COMPILER_ONLY
```
Comment (gribozavr) [inline CMakeLists.txt]: This name (outside of documentation context) can be interpreted as being related to the cross-compilation (something like "build only the compiler for the host, not for the target").

Could we use a more explicit name like `SWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER`?

File: stdlib/CMakeLists.txt [hunk 2]
```diff
@@ -1,7 +1,11 @@
 # Create convenience targets for the Swift standard library.
 
-set(CMAKE_CXX_COMPILER "${SWIFT_NATIVE_LLVM_TOOLS_PATH}/clang++")
-set(CMAKE_C_COMPILER "${SWIFT_NATIVE_LLVM_TOOLS_PATH}/clang")
+if(SWIFT_HOST_COMPILER_ONLY)
+  message(WARNING "Building swift using the host compiler, and not the just-built clang.")
```
Comment (gribozavr) [inline stdlib/CMakeLists.txt]: Building the Swift **runtime** ...

File: utils/build-script [hunk 3]
```diff
@@ -547,6 +547,10 @@ build the Debug variant of the Swift standard library and SDK overlay""",
         "--skip-test-cygwin",
         help="skip testing Swift stdlibs for Cygwin",
         action="store_true")
+    parser.add_argument(
+        "--host-compiler-only",
+        help="Use the host compiler, not the self-built one to compile swift",
```
Comment (gribozavr) [inline utils/build-script]: ... to compile the Swift **runtime**

**Pull request closed without merging**